controllers/provider-aws/pkg/infrastructure/actuator_helpers.go

```diff
@@ -0,0 +1,190 @@
+			"cidr": map[string]interface{}{
+				"worker":   infraProviderConfig.Networks.Workers[zoneIndex],
+				"public":   infraProviderConfig.Networks.Public[zoneIndex],
+				"internal": infraProviderConfig.Networks.Internal[zoneIndex],
+			},
+		})
+	}
+	return zones
+}
+
+func (c *actuator) getTerraformer(purpose, namespace, name string) (*terraformer.Terraformer, error) {
+	serverVersion, err := c.kubernetes.Discovery().ServerVersion()
+
+	if err != nil {
+		return nil, err
+	}
+	tfImage, err := awstypes.ImageVector.FindImage(awstypes.TerraformerImageName, serverVersion.GitVersion, serverVersion.GitVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	return terraformer.NewForConfig(logger.NewLogger("info"), c.config, purpose, namespace, name, tfImage.String())
+}
+
+func (c *actuator) destroyKubernetesLoadBalancersAndSecurityGroups(namespace string, tf *terraformer.Terraformer, awsClient aws.ClientInterface) error {
+	if _, err := tf.GetState(); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	vpcIDKey := "vpc_id"
+	stateVariables, err := tf.GetStateOutputVariables(vpcIDKey)
+	if err != nil {
+		if terraformer.IsVariablesNotFoundError(err) {
+			c.logger.Info("Skipping explicit AWS load balancer and security group deletion because not all variables have been found in the Terraform state.")
+			return nil
+		}
+		return err
+	}
+	vpcID := stateVariables[vpcIDKey]
+	// Find load balancers and security groups.
+	loadBalancers, err := awsClient.ListKubernetesELBs(vpcID, namespace)
+	if err != nil {
+		return err
+	}
+	securityGroups, err := awsClient.ListKubernetesSecurityGroups(vpcID, namespace)
+	if err != nil {
+		return err
+	}
+
+	// Destroy load balancers and security groups.
+	for _, loadBalancerName := range loadBalancers {
+		if err := awsClient.DeleteELB(loadBalancerName); err != nil {
+			return err
+		}
+	}
+	for _, securityGroupID := range securityGroups {
+		if err := awsClient.DeleteSecurityGroup(securityGroupID); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *actuator) injectProviderStateIntoStatus(ctx context.Context, tf *terraformer.Terraformer, infrastructure *extensionsv1alpha1.Infrastructure) error {
+	outputVarKeys := []string{
+		awstypes.VPCIDKey,
+		awstypes.SSHKeyName,
+		awstypes.IAMInstanceProfileNodes,
+		awstypes.NodesRole,
+		awstypes.SecurityGroupsNodes,
+	}
+
+	for zoneIndex := range infrastructure.Spec.Zones {
+		outputVarKeys = append(outputVarKeys, fmt.Sprintf("%s%d", awstypes.SubnetNodesPrefix, zoneIndex))
+		outputVarKeys = append(outputVarKeys, fmt.Sprintf("%s%d", awstypes.SubnetPublicPrefix, zoneIndex))
+	}
+
+	values, err := tf.GetStateOutputVariables(outputVarKeys...)
+	if err != nil {
+		return err
+	}
+
+	var (
+		instanceProfiles = []v1alpha1.InstanceProfile{
+			{
+				Purpose: "instance profile for nodes",
+				Name:    values[awstypes.IAMInstanceProfileNodes],
+			},
+		}
+		roles = []v1alpha1.Role{
+			{
+				Purpose: "role for nodes",
```

Comment:
I guess we need to define some constants for that, let's call it simply `nodes` (should only be one word)